src/eduid_webapp/eidas/acs_actions.py

```diff
@@ -188,3 +189,56 @@ def nin_verify_action(session_info, user):
     new_query_string = urlencode({'msg': 'eidas.nin_verify_success'})
     url = urlunsplit((scheme, netloc, path, new_query_string, fragment))
     return redirect(url)
+
+
+@acs_action('mfa-authentication-action')
+@require_user
+def mfa_authentication_action(session_info, user):
+    relay_state = request.form.get('RelayState')
+    current_app.logger.debug('RelayState: {}'.format(relay_state))
+    redirect_url = session.pop(relay_state, None)
+    if not redirect_url:
+        # With no redirect url just redirect the user to dashboard for a new try to log in
+        scheme, netloc, path, query_string, fragment = urlsplit(current_app.config['DASHBOARD_URL'])
```

Comment:
Log error?

Reply:
Also, is the dashboard a sensible default? The user might have been trying to log in to any SP, and suddenly ends up in our dashboard?